hibernate-core/src/main/java/org/hibernate/type/descriptor/java/DoubleJavaType.java

```diff
@@ -151,9 +156,14 @@ public long getDefaultSqlLength(Dialect dialect, JdbcType jdbcType) {
 
 	@Override
 	public int getDefaultSqlPrecision(Dialect dialect, JdbcType jdbcType) {
-		//this is the number of *binary* digits
-		//in a double-precision FP number
-		return dialect.getDoublePrecision();
+		if ( jdbcType.isFloat() ) {
+			//this is the number of *binary* digits
+			//in a double-precision FP number
+			return dialect.getDoublePrecision();
+		}
+		else {
+			return Math.min( dialect.getDefaultDecimalPrecision(), (int) ceil( dialect.getDoublePrecision() * LOG_BASE10OF2 ) );
```

Comment:
@beikov I don't understand this logic, and it looks wrong to me. Why would `dialect.getDoublePrecision()` come into it if we're not using a floating-point type on the database side? Note that `dialect.getDoublePrecision()` returns a value that might be measured in binary or decimal digits depending on the database, and so multiplying it by `log_10(2)` is probably wrong. It's only meant as an argument to the `float(n)` type.

Perhaps this should be the number of decimal digits of precision in a *Java* `double`, i.e. 17?